internal/worker/server.go

```diff
@@ -817,6 +818,30 @@ func (s *Server) requestJob(ctx context.Context, arch string, jobTypes []string,
 	return
 }
 
+func (s *Server) UpdateJobResult(token uuid.UUID, partial json.RawMessage) error {
+	jobId, err := s.jobs.IdFromToken(token)
+	if err != nil {
+		switch err {
+		case jobqueue.ErrNotExist:
+			return ErrInvalidToken
+		default:
+			return err
+		}
+	}
+
+	err = s.jobs.UpdateJobResult(jobId, partial)
+	if err != nil {
+		switch err {
+		case jobqueue.ErrNotRunning:
+			return ErrJobNotRunning
+		default:
+			return fmt.Errorf("error updating job: %w", err)
+		}
+	}
```

Comment:
This could have been just if statement and with `errors.Is`. But this is fine I guess.